components/shader/shadervisitor.cpp

```diff
@@ -250,7 +251,7 @@ namespace Shader
 
     void ShaderVisitor::createProgram(const ShaderRequirements &reqs)
     {
-        if (!reqs.mShaderRequired && !mForceShaders)
+        if (!reqs.mShaderRequired && !mForceShaders && !mAccurateFog)
```

Comment:
As I mentioned elsewhere, it's probably best to have this set up in the rendering manager.